biodata-models/src/main/java/org/opencb/biodata/models/variant/stats/VariantStats.java

```diff
@@ -436,14 +435,18 @@ public VariantStats calculate(Map<String, Map<String, String>> samplesData, Map<
                         }
                     }
 
-                    if (secondAllele < 0) {
-                        this.setMissingAlleles(this.getMissingAlleles() + 1);
-                    } else {
-                        //Allele counts for multiple alternates not supported by biodata!
-                        //Skip allele counts for genotypes like -1/2 (getAllele(1) returns 2)
-                        if (secondAllele == 0 || secondAllele == 1) {
-                            allelesCount[secondAllele]++;
-                            totalAllelesCount++;
+                    //Ensure that the genotype is not a haploid genotype with missing allele
+                    if (g.getAllelesIdx().length >= 2) {
```

Comment:
I think it would be helpful if these validations are wrapped in a private method so it is easier to understand the intention of the code.
In this case I suggest adding a small method like
```
    private boolean isHaploid(Genotype g) {
        return g.getAllelesIdx().length == 1;
    }
```
and replacing this line with 
```
if (!isHaploid(g)) {
```

I know a refactor to this code is out of the scope of the ticket but I suggest this at least for the new code we are introducing

Reply:
Actually there is a [separate part of the code](https://github.com/EBIvariation/biodata/pull/14/files/0ee966314bd2fb117aeefa3a19a88b5f34316d25#diff-e8a3bbade294ff371e0f22220224f329257d39785661adf24459ccf9552d3c6bR415) that deals with the "Haploid but not no missing allele" case. The issue that we are hitting is the "Haploid but with missing allele" case which is [unfortunately encoded](https://github.com/opencb/biodata/blob/f2e4a24a00b76ccab3c6fd1ba79dbc583ec35f00/biodata-models/src/main/java/org/opencb/biodata/models/feature/Genotype.java#L64) with "ALLELES_MISSING" regardless of haploid/diploid genotypes. Since this particular issue doesn't surface in many places, I didn't want to put in a refactored chunk of code called isHaploidWithMissingAllele() just for this special case and make it overly complicated.